internal/adapters/package.go

```diff
@@ -176,7 +186,20 @@ func (a *GenericClusterPackage) GetUnpackedHash() string {
 	return a.Status.UnpackedHash
 }
 
+func (a *GenericClusterPackage) GetSpecPaused() bool {
+	return a.Spec.Paused
+}
+
+func (a *GenericClusterPackage) SetSpecPaused(paused bool) {
+	a.Spec.Paused = paused
+}
+
 func updatePackagePhase(pkg GenericPackageAccessor) {
+	if meta.IsStatusConditionTrue(*pkg.GetConditions(), string(corev1alpha1.PackagePaused)) {
```

Comment:
I think this is not needed.
```suggestion
	if meta.IsStatusConditionTrue(*pkg.GetConditions(), corev1alpha1.PackagePaused) {
```